backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/impl/BooleanFieldSortBuilder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.types.sort.impl;
+
+import org.hibernate.search.backend.lucene.search.impl.LuceneSearchContext;
+import org.hibernate.search.backend.lucene.search.sort.impl.LuceneSearchSortCollector;
+import org.hibernate.search.backend.lucene.types.converter.impl.LuceneFieldConverter;
+import org.hibernate.search.engine.search.dsl.sort.SortOrder;
+
+import org.apache.lucene.search.SortField;
+
+public class BooleanFieldSortBuilder extends AbstractFieldSortBuilderImpl {
```

Comment:
I think you can use `IntegerFieldSortBuilder` and remove this class. You'll just need to fix `IntegerFieldSortBuilder` by replacing `LuceneFieldConverter<Integer, ?>` (which is wrong) with `LuceneFieldConverter<?, Integer>`.